wemake_python_styleguide/logic/arguments/function_args.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+import ast
+from itertools import zip_longest
+from typing import List, Optional
+
+from wemake_python_styleguide import types
+
+
+def get_args_without_self(node: types.AnyFunctionDefAndLambda) -> List[ast.arg]:
+    """Gets ``node`` arguments excluding ``self``."""
+    node_args = node.args.args
+    if not node_args:
+        return node_args
+    if isinstance(node, ast.Lambda):
+        return node_args
+    first_arg = node_args[0]
+    if first_arg.arg != 'self':
```

Comment:
What about `cls` and `mcs`?

Reply:
I will use SPECIAL_ARGUMENT_NAMES_WHITELIST setting here